src/gretel_trainer/trainer.py

```diff
@@ -0,0 +1,156 @@
+"""Main Trainer Module"""
+
+from collections import namedtuple
+from enum import Enum
+import json
+import logging
+import pandas as pd
+import pkgutil
+import yaml
+
+from gretel_client import configure_session, ClientConfig
+from gretel_client.projects import create_or_get_unique_project
+from gretel_client.projects.models import read_model_config
+from gretel_client.projects.jobs import Status
+from gretel_synthetics.utils.header_clusters import cluster
+
+from gretel_trainer import strategy, runner
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel("DEBUG")
+
+
+class ExtendedEnum(Enum):
+    """Utility class for Model enum"""
+
+    @classmethod
+    def get_types(cls):
+        return list(map(lambda c: c.name, cls))
+
+    @classmethod
+    def get_config(cls, model: str):
+        return cls[model].config
+
+
+class Model(namedtuple("Model", "config"), ExtendedEnum):
+    """Enum to pair valid models and configurations"""
+
+    GretelLSTM = "templates/gretel_lstm.yaml"
+    GretelCTGAN = "templates/gretel_ctgan.yaml"
+
+
+class Trainer:
+    """Automated generative training and sampling tool
```

Comment:
Nit: orient around "sampling" or "generation" as the term we use but we shouldn't probably mix them